pkg/neonvm/controllers/vm_controller.go

```diff
@@ -1145,23 +1162,29 @@ func affinityForVirtualMachine(vm *vmv1.VirtualMachine) *corev1.Affinity {
 	if a.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
 		a.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = &corev1.NodeSelector{}
 	}
-	// if NodeSelectorTerms list is empty - add default values (arch==amd64 or os==linux)
+
+	// if NodeSelectorTerms list is empty - add default values (arch==vm.Spec.Affinity or os==linux)
```

Comment:
question: I know it's the previous behavior, but isn't this not correct? Not sure how I'm only just noticing it, but that means we won't respect the architecture if there's any node selector terms, right?

Can we change this to always add ours?

Reply:
But didn't we decided to drop it last week? Or did I understand you incorrectly?

https://github.com/neondatabase/autoscaling/pull/1244#discussion_r1941227562

Reply:
Discussed in 1o1/DM, changed accordingly.